### PR TITLE
package vmux for PyPI

### PR DESCRIPTION
**Pull request by joshbode**
I've added a `setup.py`, `.travis.yml`, etc, and `entry_points` for packaging `vmux` so that `vmux` can be installed via `pip` from PyPI (see https://pypi.org/project/vmux/ for a provisional example from testing).

Travis will need to be enabled by @jceb, and @jceb will probbably wish to replace @joshbode's username and _encrypted_ secure password with theirs before merging :) I'm happy to be the package maintainer, if they're happy to delegate.

This may _also_ allow Windows to be supported (untested!!!)

### CODE REVIEW

**Comment by joshbode** [comment]
this should be still compatible with the existing `Makefile`

**Comment by joshbode** [comment]
additionally, I added support for `setuptools_scm` which infers the package version from the git version

**Comment by joshbode** [comment]
Travis build fixed :)

**Comment by joshbode** [comment]
@jceb - Bump :)

**Comment by jceb** [comment]
Thank you for your work. I'll have a look at it!

File: vmux/__main__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#! /usr/bin/env python3
```
Comment (jceb) [inline vmux/__main__.py]: The space shouldn't be there.

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,7 @@
 VERSION = $(shell git tag -l|tail -n 1|sed -e 's/^v//')
 DESTDIR = /usr
 SCRIPTS = scripts/nvim.vmux scripts/nvim-qt.vmux scripts/vim.vmux scripts/gvim.vmux scripts/kak.vmux
+MAIN = vmux/__main__.py
```
Comment (jceb) [inline Makefile]: Please file the changes in this file as separate pull request.
In total, please create three pull requests:
- pip inthegration
- neovim → pynvim
- compatibility with non-gnu cp

File: vmux/__main__.py (hunk 1 above)
Comment (joshbode) [reply vmux/__main__.py]: The space after a hashbang is optional, but I'll revert it.

File: Makefile (hunk 2 above)
Comment (joshbode) [reply Makefile]: will do - I'll send something through later today

Comment (joshbode) [reply Makefile]: done :)

**Comment by joshbode** [comment]
Hey @jceb - just checking in - is there anything else you want me to change for this?
I'd love to see `vmux` packaged :)

**Comment by jceb** [comment]
Thanks man, I'm sorry I didn't get to it. I'll put it on my schedule.

**Comment by joshbode** [comment]
Thanks! Please let me know if you sign up for an account on PyPI and I'll transfer the `vmux` project to you so you're the owner.

The `travis` command to generate credentials for the `.travisrc` (unless you're happy leaving my credentials there) was this:
```
$ gem install travis # or gem install --user travis
$ travis encrypt "your-pypi-password-here" --add deploy.password
```

**Comment by joshbode** [comment]
bump :)

**Comment by jceb** [comment]
Thanks mate. I'm sorry that it took so long.
Yes, please transfer the `vmux` project to me.

**Comment by jceb** [comment]
My pypi user name is `jceb81`

**Comment by joshbode** [comment]
No worries! I've just added you as owner

**Comment by jceb** [comment]
Package is deployed, enjoy :-D

**Comment by joshbode** [comment]
Awesome! Thank you for that - I can get rid of my local install now :)

**Comment by joshbode** [comment]
Ta! It works :)
```
$ pipx install vmux
  installed package vmux 1.1.dev14, Python 3.8.3
  These apps are now globally available
    - vmux
done! ✨ 🌟 ✨
```

**Pull request merged**